### ISSUE TITLE
GFN1-xTB non-convergence with biological ligands

### ISSUE THREAD

**Issue by linminhtoo** [tblite/tblite#103]
hello,

I've been using `tblite` with great success on many small molecules (1-30 atoms). I am moving on to try it on larger biologically active ligand (drug) molecules, which have more atoms (30-150). 

However, I am starting to notice `GFN1-xTB` **fails to converge** for several molecules. I've also tried `GFN2-xTB`, and it also fails to converge.

Might it be because many of these molecules are not in their equilibrium position? Due to their interaction with the protein molecule they're binding to, some of their bonds would be slightly distorted from the equilibrium position. Some of these are from the open-sourced `QMugs` dataset, which has 2 million+ drug-like molecules. 
On a 3000 molecule subset from `QMugs`, at least 2% failed to converge. It's small, but non-trivial.

It should also be noted that these molecules could be converged with both Hartree-Fock & DFT calculations.

Any help or advice would be greatly appreciated! :) 

Here is an example code snippet that I'm running:
```python

calc = tb.Calculator("GFN1-xTB", atomic_nums, xyzs_bohr, charge=charge)
calc.set("save-integrals", 1)  # 1 = yes, save integrals, default = 0 (False)
calc.set("accuracy", 0.1)  # lowered from the default of 1
calc.set("guess", 0)
calc.set("max-iter", 150)
calc.set(
    "mixer-damping", 0.4
)  # default 0.4, not sure what this changes... if too low seems to increase no. of iterations.
calc.set("temperature", 300 * 3.166808578545117e-6)  # in Hartrees, not Kelvins.
# calc.set("verbosity", 0)  # STILL not supported in v0.3.0 (latest conda release)...

res = calc.singlepoint()
res_dict = res.dict()  # energy is in atomic units, aka Hartrees
```

Here's an example molecule with 42 atoms that fails to converge within 150 iterations:
```python
atomic_nums = np.array([6, 6, 6, 6, 6, 8, 6, 6, 6, 7, 6, 7, 7, 6, 6, 6, 8, 6, 6, 6, 6, 6,
       6, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
charge = 0
xyzs_bohr = np.array([[-63.91676791, 329.07330697, 157.05191297],
       [-64.42548096, 326.63724556, 158.53288941],
       [-62.60492276, 324.4927918 , 157.81498615],
       [-63.14311232, 322.01099818, 159.22736428],
       [-61.17987888, 319.97406988, 158.56426181],
       [-63.08056214, 322.52671091, 161.8739351 ],
       [-65.34899169, 323.22232477, 162.93350345],
       [-67.37837245, 321.52723456, 163.39269971],
       [-69.6193944 , 322.48873531, 164.52275505],
       [-69.99261932, 324.9365037 , 165.19776635],
       [-68.0681184 , 326.58661142, 164.714565  ],
       [-67.98723642, 329.06102329, 165.17376127],
       [-65.60410873, 329.92122693, 164.34210421],
       [-64.28696516, 327.96552662, 163.39894248],
       [-65.79950143, 325.76513309, 163.59982102],
       [-67.2551608 , 318.82303713, 162.64493789],
       [-66.058021  , 318.15899752, 160.76465534],
       [-68.62653566, 316.96638481, 164.26518703],
       [-69.88716998, 317.82298408, 166.46010192],
       [-71.21375699, 316.11601388, 168.01270433],
       [-71.30352012, 313.5342784 , 167.38172397],
       [-70.07123054, 312.66216593, 165.19284999],
       [-68.74369919, 314.36688701, 163.63874817],
       [-65.23334199, 330.55709481, 157.60844154],
       [-64.12993013, 328.77244331, 155.02480296],
       [-62.00739092, 329.76756559, 157.39451519],
       [-64.26995976, 327.03332216, 160.54904215],
       [-66.36982031, 326.03724833, 158.20445955],
       [-62.74551467, 324.15718204, 155.78617487],
       [-60.66228467, 325.08408088, 158.17026132],
       [-65.00090362, 321.32769684, 158.65516393],
       [-61.13414668, 319.62522489, 156.53431156],
       [-59.28807317, 320.54456897, 159.14780871],
       [-61.61111247, 318.18033537, 159.48021779],
       [-71.19580724, 321.24133623, 164.88425857],
       [-71.65765632, 325.43595353, 165.98634402],
       [-62.39364923, 328.13297106, 162.64078566],
       [-69.84276418, 319.79889768, 166.97562722],
       [-72.16485553, 316.7898862 , 169.68851838],
       [-72.32264748, 312.22600878, 168.57205872],
       [-70.14908486, 310.68247496, 164.70076749],
       [-67.83606209, 313.64650395, 161.95783033]])
```

**Comment by linminhtoo** [comment]
Here's the STDOUT for the above molecule:
```
------------------------------------------------------------
  cycle        total energy    energy error   density error
------------------------------------------------------------
      1     -5632846.875020  -5.6328535E+06   1.3211270E+01
      2     -61730899.21241  -5.6098052E+07   1.8232228E+01
      3     -71344059.29718  -9.6131601E+06   1.7218340E+01
      4     -71294473.04956   4.9586248E+04   1.6302113E+01
      5     -73237953.21895  -1.9434802E+06   1.4347425E+01
      6     -72746739.81278   4.9121341E+05   1.2804348E+01
      7     -73157346.70792  -4.1060690E+05   1.0066934E+01
      8     -73254167.11110  -9.6820403E+04   8.8253076E+00
      9     -72793241.22093   4.6092589E+05   6.4724157E+00
     10     -71427236.41170   1.3660048E+06   6.2758152E+00
     11     -72246375.32714  -8.1913892E+05   6.1352267E+00
     12     -72793768.79141  -5.4739346E+05   5.7389370E+00
     13     -72672856.62197   1.2091217E+05   5.7614357E+00
     14     -71366122.34251   1.3067343E+06   5.2919786E+00
     15     -72440915.45663  -1.0747931E+06   3.8324271E+00
     16     -71988703.73830   4.5221172E+05   4.0002133E+00
     17     -71522438.76538   4.6626497E+05   4.0976385E+00
     18     -72678289.67316  -1.1558509E+06   3.9716528E+00
     19     -72497022.12515   1.8126755E+05   3.8952833E+00
     20     -72033324.09697   4.6369803E+05   3.8326879E+00
     21     -71393254.86746   6.4006923E+05   3.9298418E+00
     22     -72558019.81385  -1.1647649E+06   3.7056984E+00
     23     -72379389.70481   1.7863011E+05   3.6522341E+00
     24     -72656326.68983  -2.7693699E+05   3.6684827E+00
     25     -72566436.22940   8.9890460E+04   3.7802547E+00
     26     -73158270.60993  -5.9183438E+05   3.6236987E+00
     27     -74153603.62574  -9.9533302E+05   3.4736314E+00
     28     -73961915.69359   1.9168793E+05   3.3040564E+00
     29     -73340138.95466   6.2177674E+05   3.6631612E+00
     30     -73103485.43962   2.3665352E+05   3.8866150E+00
     31     -72451794.62331   6.5169082E+05   3.3117110E+00
     32     -73926100.29076  -1.4743057E+06   3.3829858E+00
     33     -73365832.98633   5.6026730E+05   3.3297880E+00
     34     -73968249.64410  -6.0241666E+05   3.2816720E+00
     35     -73693396.80559   2.7485284E+05   3.3352122E+00
     36     -72881934.10947   8.1146270E+05   4.0086118E+00
     37     -74001118.82601  -1.1191847E+06   3.3047349E+00
     38     -74241649.74076  -2.4053091E+05   2.7202961E+00
     39     -73586575.25784   6.5507448E+05   2.8435656E+00
     40     -73885815.86720  -2.9924061E+05   3.1404747E+00
     41     -73203006.29909   6.8280957E+05   3.1710618E+00
     42     -73342304.76236  -1.3929846E+05   3.0918959E+00
     43     -74205459.06257  -8.6315430E+05   3.3405796E+00
     44     -73872590.97582   3.3286809E+05   3.5527001E+00
     45     -73668636.38919   2.0395459E+05   3.6095407E+00
     46     -72237992.74964   1.4306436E+06   3.8764181E+00
     47     -71709518.77398   5.2847398E+05   3.9973382E+00
     48     -73660163.34140  -1.9506446E+06   3.6919989E+00
     49     -72986454.19312   6.7370915E+05   3.4443007E+00
     50     -71110490.94209   1.8759633E+06   3.8315516E+00
     51     -70182100.88004   9.2839006E+05   4.0842302E+00
     52     -72307261.61729  -2.1251607E+06   3.4776190E+00
     53     -73264350.78248  -9.5708917E+05   3.2395967E+00
     54     -73384047.55419  -1.1969677E+05   3.3835748E+00
     55     -71616691.30955   1.7673562E+06   3.5295219E+00
     56     -71912168.88627  -2.9547758E+05   3.6528380E+00
     57     -73487444.67341  -1.5752758E+06   3.2347100E+00
     58     -73242392.15139   2.4505252E+05   3.1283033E+00
     59     -73532626.41293  -2.9023426E+05   3.2009485E+00
     60     -71677502.81086   1.8551236E+06   3.8420703E+00
     61     -71449386.33764   2.2811647E+05   3.5687804E+00
     62     -72265242.60659  -8.1585627E+05   3.6308243E+00
     63     -71968151.43847   2.9709117E+05   3.8584144E+00
     64     -71559688.09850   4.0846334E+05   3.7256262E+00
     65     -71849824.92681  -2.9013683E+05   3.9839656E+00
     66     -72406367.95731  -5.5654303E+05   3.7322606E+00
     67     -71914914.96940   4.9145299E+05   3.5890944E+00
     68     -74181246.17944  -2.2663312E+06   2.8855163E+00
     69     -74225946.25537  -4.4700076E+04   2.5366546E+00
     70     -74008093.99307   2.1785226E+05   2.3706790E+00
     71     -71892640.91059   2.1154531E+06   2.4591730E+00
     72     -72313412.59621  -4.2077169E+05   2.3009248E+00
     73     -73973534.60175  -1.6601220E+06   1.9662215E+00
     74     -72934424.02450   1.0391106E+06   2.2361663E+00
     75     -72824124.14618   1.1029988E+05   2.2748525E+00
     76     -74008787.31574  -1.1846632E+06   2.2287709E+00
     77     -73645161.56353   3.6362575E+05   1.9670120E+00
     78     -72360000.07137   1.2851615E+06   2.4219951E+00
     79     -72622921.95826  -2.6292189E+05   2.1659147E+00
     80     -73416960.40621  -7.9403845E+05   2.3653999E+00
     81     -73558773.54043  -1.4181313E+05   2.2043175E+00
     82     -71853074.08948   1.7056995E+06   2.3383746E+00
     83     -72415205.70423  -5.6213161E+05   2.1693443E+00
     84     -74039175.90832  -1.6239702E+06   2.3131171E+00
     85     -73909231.40472   1.2994450E+05   2.3831921E+00
     86     -73520918.85423   3.8831255E+05   2.0482634E+00
     87     -73776407.55022  -2.5548870E+05   2.3074620E+00
     88     -74014106.11282  -2.3769856E+05   2.2641564E+00
     89     -73810622.89008   2.0348322E+05   1.9389637E+00
     90     -73238478.33608   5.7214455E+05   1.9046114E+00
     91     -73751437.70755  -5.1295937E+05   2.0984251E+00
     92     -74098227.40236  -3.4678969E+05   2.0929910E+00
     93     -73892558.49526   2.0566891E+05   1.8748612E+00
     94     -72187399.48030   1.7051590E+06   1.5552392E+00
     95     -73926530.85651  -1.7391314E+06   1.9699951E+00
     96     -74092554.59160  -1.6602374E+05   1.9956406E+00
     97     -72806417.58302   1.2861370E+06   1.6938921E+00
     98     -72561270.76365   2.4514682E+05   1.5441275E+00
     99     -73953735.21997  -1.3924645E+06   2.1499966E+00
    100     -72780296.42969   1.1734388E+06   1.8970346E+00
    101     -71860735.05341   9.1956138E+05   1.4249771E+00
    102     -74162312.54553  -2.3015775E+06   2.0123602E+00
    103     -74201444.37724  -3.9131832E+04   2.1307916E+00
    104     -74179398.36343   2.2046014E+04   1.9482029E+00
    105     -74084621.97538   9.4776388E+04   2.1439947E+00
    106     -74120662.59922  -3.6040624E+04   2.0052384E+00
    107     -74332139.69225  -2.1147709E+05   2.1144492E+00
    108     -73375318.49680   9.5682120E+05   2.3076460E+00
    109     -73364399.54747   1.0918949E+04   2.1245481E+00
    110     -74134894.09193  -7.7049454E+05   1.9249116E+00
    111     -73102720.05549   1.0321740E+06   1.7513225E+00
    112     -72356026.71368   7.4669334E+05   1.4479316E+00
    113     -74175322.56329  -1.8192958E+06   1.7274044E+00
    114     -74048169.94431   1.2715262E+05   1.4767306E+00
    115     -74342110.68366  -2.9394074E+05   1.7258816E+00
    116     -74259767.69508   8.2342989E+04   1.6505724E+00
    117     -74304287.45794  -4.4519763E+04   1.6074933E+00
    118     -74129412.43370   1.7487502E+05   1.9796980E+00
    119     -73958692.50940   1.7071992E+05   1.4741236E+00
    120     -73878568.17389   8.0124336E+04   1.2004283E+00
    121     -73642173.02024   2.3639515E+05   2.1344349E+00
    122     -74353713.15823  -7.1154014E+05   1.5528711E+00
    123     -74162669.01320   1.9104415E+05   2.0403045E+00
    124     -74316966.97379  -1.5429796E+05   1.5554777E+00
    125     -73969076.26379   3.4789071E+05   1.5163782E+00
    126     -73635489.43176   3.3358683E+05   1.4935952E+00
    127     -74271955.54014  -6.3646611E+05   1.5683303E+00
    128     -72591909.71115   1.6800458E+06   1.4105309E+00
    129     -74209532.67619  -1.6176230E+06   1.8035914E+00
    130     -73258184.90073   9.5134778E+05   1.6887729E+00
    131     -73994354.81561  -7.3616991E+05   1.0869238E+00
    132     -74303252.65124  -3.0889784E+05   1.5673986E+00
    133     -74164475.19498   1.3877746E+05   1.5064444E+00
    134     -73723301.19263   4.4117400E+05   2.0933021E+00
    135     -74167529.72861  -4.4422854E+05   1.5111769E+00
    136     -73601494.01116   5.6603572E+05   1.3841924E+00
    137     -74287514.62794  -6.8602062E+05   1.4556588E+00
    138     -74261807.51295   2.5707115E+04   1.1013120E+00
    139     -74195730.01956   6.6077493E+04   1.6120390E+00
    140     -74295293.33944  -9.9563320E+04   1.5793481E+00
    141     -74250590.60844   4.4702731E+04   1.3804818E+00
    142     -73710963.44371   5.3962716E+05   9.5025551E-01
    143     -74305195.87117  -5.9423243E+05   1.6217243E+00
    144     -74026675.12981   2.7852074E+05   1.9869490E+00
    145     -74121300.59261  -9.4625463E+04   1.4301677E+00
    146     -74214861.20270  -9.3560610E+04   1.7252856E+00
    147     -74035005.04279   1.7985616E+05   2.2149646E+00
    148     -73942598.06855   9.2406974E+04   1.2153511E+00
    149     -73178056.24769   7.6454182E+05   2.4039385E+00
    150     -74050372.31586  -8.7231607E+05   1.4605443E+00
------------------------------------------------------------

 total:                                  32.390 sec
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[37], line 1
----> 1 outs = calc_singlepoint_xtb_for_molecule(
      2     atomic_nums=np.array(frag_atom_nums),
      3     xyzs_bohr=frag_xyzs,
      4     charge=frag_charge,
      5     shells_map=GFN1_XTB_SHELLS_MAP,
      6     accuracy=0.1,
      7     xtb_method="GFN1-xTB",
      8     max_iter=150
      9 )

File ~/machineer-nix/repos/orbnet2/notebooks/tblite/../../src/orbnet2/interface/gen_features_with_tblite.py:123, in calc_singlepoint_xtb_for_molecule(atomic_nums, xyzs_bohr, charge, shells_map, xtb_method, accuracy, guess_mode, max_iter, temp)
    120 calc.set("temperature", temp * 3.166808578545117e-6)  # in Hartrees, not Kelvins.
    121 # calc.set("verbosity", 0)  # STILL not supported in v0.3.0 (latest conda release)...
--> 123 res = calc.singlepoint()
    124 res_dict = res.dict()  # energy is in atomic units, aka Hartrees
    126 new_order = get_new_order(atomic_nums, shells_map)

File /nix/store/81s1qa2yjwi0av1a307asg02qxrfcbb3-python3-3.10.9-env/lib/python3.10/site-packages/tblite/interface.py:522, in Calculator.singlepoint(self, res, copy)
    506 """
    507 Perform actual single point calculation in the library backend.
    508 The output of the library will be forwarded to the standard output.
   (...)
    517     in case the calculation fails
    518 """
    520 _res = Result(res) if copy or res is None else res
--> 522 library.get_singlepoint(self._ctx, self._mol, self._calc, _res._res)
    523 return _res

File /nix/store/81s1qa2yjwi0av1a307asg02qxrfcbb3-python3-3.10.9-env/lib/python3.10/site-packages/tblite/library.py:95, in context_check.<locals>.handle_context_error(ctx, *args, **kwargs)
     93     _message = ffi.new("char[]", 512)
     94     lib.tblite_get_context_error(ctx, _message, ffi.NULL)
---> 95     raise RuntimeError(ffi.string(_message).decode())
     96 return value

RuntimeError: SCF not converged in 150 cycles
```

**Comment by linminhtoo** [comment]
Here's the STDOUT for the above molecule:
```
------------------------------------------------------------
  cycle        total energy    energy error   density error
------------------------------------------------------------
      1     -5632846.875020  -5.6328535E+06   1.3211270E+01
      2     -61730899.21241  -5.6098052E+07   1.8232228E+01
      3     -71344059.29718  -9.6131601E+06   1.7218340E+01
      4     -71294473.04956   4.9586248E+04   1.6302113E+01
      5     -73237953.21895  -1.9434802E+06   1.4347425E+01
      6     -72746739.81278   4.9121341E+05   1.2804348E+01
      7     -73157346.70792  -4.1060690E+05   1.0066934E+01
      8     -73254167.11110  -9.6820403E+04   8.8253076E+00
      9     -72793241.22093   4.6092589E+05   6.4724157E+00
     10     -71427236.41170   1.3660048E+06   6.2758152E+00
     11     -72246375.32714  -8.1913892E+05   6.1352267E+00
     12     -72793768.79141  -5.4739346E+05   5.7389370E+00
     13     -72672856.62197   1.2091217E+05   5.7614357E+00
     14     -71366122.34251   1.3067343E+06   5.2919786E+00
     15     -72440915.45663  -1.0747931E+06   3.8324271E+00
     16     -71988703.73830   4.5221172E+05   4.0002133E+00
     17     -71522438.76538   4.6626497E+05   4.0976385E+00
     18     -72678289.67316  -1.1558509E+06   3.9716528E+00
     19     -72497022.12515   1.8126755E+05   3.8952833E+00
     20     -72033324.09697   4.6369803E+05   3.8326879E+00
     21     -71393254.86746   6.4006923E+05   3.9298418E+00
     22     -72558019.81385  -1.1647649E+06   3.7056984E+00
     23     -72379389.70481   1.7863011E+05   3.6522341E+00
     24     -72656326.68983  -2.7693699E+05   3.6684827E+00
     25     -72566436.22940   8.9890460E+04   3.7802547E+00
     26     -73158270.60993  -5.9183438E+05   3.6236987E+00
     27     -74153603.62574  -9.9533302E+05   3.4736314E+00
     28     -73961915.69359   1.9168793E+05   3.3040564E+00
     29     -73340138.95466   6.2177674E+05   3.6631612E+00
     30     -73103485.43962   2.3665352E+05   3.8866150E+00
     31     -72451794.62331   6.5169082E+05   3.3117110E+00
     32     -73926100.29076  -1.4743057E+06   3.3829858E+00
     33     -73365832.98633   5.6026730E+05   3.3297880E+00
     34     -73968249.64410  -6.0241666E+05   3.2816720E+00
     35     -73693396.80559   2.7485284E+05   3.3352122E+00
     36     -72881934.10947   8.1146270E+05   4.0086118E+00
     37     -74001118.82601  -1.1191847E+06   3.3047349E+00
     38     -74241649.74076  -2.4053091E+05   2.7202961E+00
     39     -73586575.25784   6.5507448E+05   2.8435656E+00
     40     -73885815.86720  -2.9924061E+05   3.1404747E+00
     41     -73203006.29909   6.8280957E+05   3.1710618E+00
     42     -73342304.76236  -1.3929846E+05   3.0918959E+00
     43     -74205459.06257  -8.6315430E+05   3.3405796E+00
     44     -73872590.97582   3.3286809E+05   3.5527001E+00
     45     -73668636.38919   2.0395459E+05   3.6095407E+00
     46     -72237992.74964   1.4306436E+06   3.8764181E+00
     47     -71709518.77398   5.2847398E+05   3.9973382E+00
     48     -73660163.34140  -1.9506446E+06   3.6919989E+00
     49     -72986454.19312   6.7370915E+05   3.4443007E+00
     50     -71110490.94209   1.8759633E+06   3.8315516E+00
     51     -70182100.88004   9.2839006E+05   4.0842302E+00
     52     -72307261.61729  -2.1251607E+06   3.4776190E+00
     53     -73264350.78248  -9.5708917E+05   3.2395967E+00
     54     -73384047.55419  -1.1969677E+05   3.3835748E+00
     55     -71616691.30955   1.7673562E+06   3.5295219E+00
     56     -71912168.88627  -2.9547758E+05   3.6528380E+00
     57     -73487444.67341  -1.5752758E+06   3.2347100E+00
     58     -73242392.15139   2.4505252E+05   3.1283033E+00
     59     -73532626.41293  -2.9023426E+05   3.2009485E+00
     60     -71677502.81086   1.8551236E+06   3.8420703E+00
     61     -71449386.33764   2.2811647E+05   3.5687804E+00
     62     -72265242.60659  -8.1585627E+05   3.6308243E+00
     63     -71968151.43847   2.9709117E+05   3.8584144E+00
     64     -71559688.09850   4.0846334E+05   3.7256262E+00
     65     -71849824.92681  -2.9013683E+05   3.9839656E+00
     66     -72406367.95731  -5.5654303E+05   3.7322606E+00
     67     -71914914.96940   4.9145299E+05   3.5890944E+00
     68     -74181246.17944  -2.2663312E+06   2.8855163E+00
     69     -74225946.25537  -4.4700076E+04   2.5366546E+00
     70     -74008093.99307   2.1785226E+05   2.3706790E+00
     71     -71892640.91059   2.1154531E+06   2.4591730E+00
     72     -72313412.59621  -4.2077169E+05   2.3009248E+00
     73     -73973534.60175  -1.6601220E+06   1.9662215E+00
     74     -72934424.02450   1.0391106E+06   2.2361663E+00
     75     -72824124.14618   1.1029988E+05   2.2748525E+00
     76     -74008787.31574  -1.1846632E+06   2.2287709E+00
     77     -73645161.56353   3.6362575E+05   1.9670120E+00
     78     -72360000.07137   1.2851615E+06   2.4219951E+00
     79     -72622921.95826  -2.6292189E+05   2.1659147E+00
     80     -73416960.40621  -7.9403845E+05   2.3653999E+00
     81     -73558773.54043  -1.4181313E+05   2.2043175E+00
     82     -71853074.08948   1.7056995E+06   2.3383746E+00
     83     -72415205.70423  -5.6213161E+05   2.1693443E+00
     84     -74039175.90832  -1.6239702E+06   2.3131171E+00
     85     -73909231.40472   1.2994450E+05   2.3831921E+00
     86     -73520918.85423   3.8831255E+05   2.0482634E+00
     87     -73776407.55022  -2.5548870E+05   2.3074620E+00
     88     -74014106.11282  -2.3769856E+05   2.2641564E+00
     89     -73810622.89008   2.0348322E+05   1.9389637E+00
     90     -73238478.33608   5.7214455E+05   1.9046114E+00
     91     -73751437.70755  -5.1295937E+05   2.0984251E+00
     92     -74098227.40236  -3.4678969E+05   2.0929910E+00
     93     -73892558.49526   2.0566891E+05   1.8748612E+00
     94     -72187399.48030   1.7051590E+06   1.5552392E+00
     95     -73926530.85651  -1.7391314E+06   1.9699951E+00
     96     -74092554.59160  -1.6602374E+05   1.9956406E+00
     97     -72806417.58302   1.2861370E+06   1.6938921E+00
     98     -72561270.76365   2.4514682E+05   1.5441275E+00
     99     -73953735.21997  -1.3924645E+06   2.1499966E+00
    100     -72780296.42969   1.1734388E+06   1.8970346E+00
    101     -71860735.05341   9.1956138E+05   1.4249771E+00
    102     -74162312.54553  -2.3015775E+06   2.0123602E+00
    103     -74201444.37724  -3.9131832E+04   2.1307916E+00
    104     -74179398.36343   2.2046014E+04   1.9482029E+00
    105     -74084621.97538   9.4776388E+04   2.1439947E+00
    106     -74120662.59922  -3.6040624E+04   2.0052384E+00
    107     -74332139.69225  -2.1147709E+05   2.1144492E+00
    108     -73375318.49680   9.5682120E+05   2.3076460E+00
    109     -73364399.54747   1.0918949E+04   2.1245481E+00
    110     -74134894.09193  -7.7049454E+05   1.9249116E+00
    111     -73102720.05549   1.0321740E+06   1.7513225E+00
    112     -72356026.71368   7.4669334E+05   1.4479316E+00
    113     -74175322.56329  -1.8192958E+06   1.7274044E+00
    114     -74048169.94431   1.2715262E+05   1.4767306E+00
    115     -74342110.68366  -2.9394074E+05   1.7258816E+00
    116     -74259767.69508   8.2342989E+04   1.6505724E+00
    117     -74304287.45794  -4.4519763E+04   1.6074933E+00
    118     -74129412.43370   1.7487502E+05   1.9796980E+00
    119     -73958692.50940   1.7071992E+05   1.4741236E+00
    120     -73878568.17389   8.0124336E+04   1.2004283E+00
    121     -73642173.02024   2.3639515E+05   2.1344349E+00
    122     -74353713.15823  -7.1154014E+05   1.5528711E+00
    123     -74162669.01320   1.9104415E+05   2.0403045E+00
    124     -74316966.97379  -1.5429796E+05   1.5554777E+00
    125     -73969076.26379   3.4789071E+05   1.5163782E+00
    126     -73635489.43176   3.3358683E+05   1.4935952E+00
    127     -74271955.54014  -6.3646611E+05   1.5683303E+00
    128     -72591909.71115   1.6800458E+06   1.4105309E+00
    129     -74209532.67619  -1.6176230E+06   1.8035914E+00
    130     -73258184.90073   9.5134778E+05   1.6887729E+00
    131     -73994354.81561  -7.3616991E+05   1.0869238E+00
    132     -74303252.65124  -3.0889784E+05   1.5673986E+00
    133     -74164475.19498   1.3877746E+05   1.5064444E+00
    134     -73723301.19263   4.4117400E+05   2.0933021E+00
    135     -74167529.72861  -4.4422854E+05   1.5111769E+00
    136     -73601494.01116   5.6603572E+05   1.3841924E+00
    137     -74287514.62794  -6.8602062E+05   1.4556588E+00
    138     -74261807.51295   2.5707115E+04   1.1013120E+00
    139     -74195730.01956   6.6077493E+04   1.6120390E+00
    140     -74295293.33944  -9.9563320E+04   1.5793481E+00
    141     -74250590.60844   4.4702731E+04   1.3804818E+00
    142     -73710963.44371   5.3962716E+05   9.5025551E-01
    143     -74305195.87117  -5.9423243E+05   1.6217243E+00
    144     -74026675.12981   2.7852074E+05   1.9869490E+00
    145     -74121300.59261  -9.4625463E+04   1.4301677E+00
    146     -74214861.20270  -9.3560610E+04   1.7252856E+00
    147     -74035005.04279   1.7985616E+05   2.2149646E+00
    148     -73942598.06855   9.2406974E+04   1.2153511E+00
    149     -73178056.24769   7.6454182E+05   2.4039385E+00
    150     -74050372.31586  -8.7231607E+05   1.4605443E+00
------------------------------------------------------------

 total:                                  32.390 sec
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[37], line 1
----> 1 outs = calc_singlepoint_xtb_for_molecule(
      2     atomic_nums=np.array(frag_atom_nums),
      3     xyzs_bohr=frag_xyzs,
      4     charge=frag_charge,
      5     shells_map=GFN1_XTB_SHELLS_MAP,
      6     accuracy=0.1,
      7     xtb_method="GFN1-xTB",
      8     max_iter=150
      9 )

calc_singlepoint_xtb_for_molecule(atomic_nums, xyzs_bohr, charge, shells_map, xtb_method, accuracy, guess_mode, max_iter, temp)
    120 calc.set("temperature", temp * 3.166808578545117e-6)  # in Hartrees, not Kelvins.
    121 # calc.set("verbosity", 0)  # STILL not supported in v0.3.0 (latest conda release)...
--> 123 res = calc.singlepoint()
    124 res_dict = res.dict()  # energy is in atomic units, aka Hartrees
    126 new_order = get_new_order(atomic_nums, shells_map)

File /nix/store/81s1qa2yjwi0av1a307asg02qxrfcbb3-python3-3.10.9-env/lib/python3.10/site-packages/tblite/interface.py:522, in Calculator.singlepoint(self, res, copy)
    506 """
    507 Perform actual single point calculation in the library backend.
    508 The output of the library will be forwarded to the standard output.
   (...)
    517     in case the calculation fails
    518 """
    520 _res = Result(res) if copy or res is None else res
--> 522 library.get_singlepoint(self._ctx, self._mol, self._calc, _res._res)
    523 return _res

File /nix/store/81s1qa2yjwi0av1a307asg02qxrfcbb3-python3-3.10.9-env/lib/python3.10/site-packages/tblite/library.py:95, in context_check.<locals>.handle_context_error(ctx, *args, **kwargs)
     93     _message = ffi.new("char[]", 512)
     94     lib.tblite_get_context_error(ctx, _message, ffi.NULL)
---> 95     raise RuntimeError(ffi.string(_message).decode())
     96 return value

RuntimeError: SCF not converged in 150 cycles
```

**Comment by marvinfriede** [comment]
Hi, I cannot reproduce your issue. Both the Fortran implementation and the Python API give the same results for me. I used version 0.3.0 for both and Python 3.9.

<details>
  <summary><code>tblite run --method gfn1 coord</code></summary>

  ```
xTB calculator
 | screened Coulomb repulsion
 | Coulomb electrostatics
 |  * isotropic Klopman-Ohno electrostatics
 |  * onsite third-order electrostatics
 | DFT-D3(BJ) dispersion correction

halogen bonding energy    0.0000000000000E+00 Eh
repulsion energy          8.5419510371271E-01 Eh
dispersion energy        -2.9125299376037E-02 Eh
number of electrons       1.1800000000000E+02 e
integral cutoff           1.9731147476801E+01 bohr

------------------------------------------------------------
  cycle        total energy    energy error   density error
------------------------------------------------------------
      1     -66.55442482087  -6.7379495E+01   6.7607540E-01
      2     -67.02432320634  -4.6989839E-01   4.0545846E-01
      3     -66.90920734717   1.1511586E-01   1.9286613E-01
      4     -67.15202983501  -2.4282249E-01   3.6717101E-02
      5     -67.16473094557  -1.2701111E-02   9.5444080E-03
      6     -67.16597839198  -1.2474464E-03   2.1005103E-03
      7     -67.16599923701  -2.0845023E-05   9.9988460E-04
      8     -67.16600812617  -8.8891622E-06   2.8875558E-04
      9     -67.16600829862  -1.7245404E-07   1.3878143E-04
     10     -67.16600833001  -3.1385852E-08   5.0661537E-05
     11     -67.16600834372  -1.3715024E-08   1.1852910E-05
------------------------------------------------------------

electronic energy        -6.7991078148060E+01 Eh
total energy             -6.7166008343723E+01 Eh

 total:                                   0.098 sec
 - repulsion                              0.001 sec (  0%)
 - halogen                                0.000 sec (  0%)
 - dispersion                             0.000 sec (  0%)
 - coulomb                                0.000 sec (  0%)
 - hamiltonian                            0.010 sec (  9%)
 - scc                                    0.087 sec ( 88%)

  ```
</details>

<details>
  <summary><code>Python API</code></summary>

  ```
------------------------------------------------------------
  cycle        total energy    energy error   density error
------------------------------------------------------------
      1     -66.55442482087  -6.7379495E+01   6.7607540E-01
      2     -67.02432320634  -4.6989839E-01   4.0545846E-01
      3     -66.90920734717   1.1511586E-01   1.9286613E-01
      4     -67.15202983501  -2.4282249E-01   3.6717101E-02
      5     -67.16473094557  -1.2701111E-02   9.5444080E-03
      6     -67.16597839198  -1.2474464E-03   2.1005103E-03
      7     -67.16599923701  -2.0845023E-05   9.9988460E-04
      8     -67.16600812617  -8.8891622E-06   2.8875558E-04
      9     -67.16600829862  -1.7245401E-07   1.3878143E-04
     10     -67.16600833001  -3.1385881E-08   5.0661537E-05
     11     -67.16600834372  -1.3715066E-08   1.1852910E-05
     12     -67.16600834440  -6.8034467E-10   5.5164728E-06
     13     -67.16600834456  -1.5231194E-10   8.2998135E-07
------------------------------------------------------------

 total:                                   0.101 sec

  ```
</details>

Considering the energy is way too high in the first step, this is not a convergence issue. Have you tested the Fortran version?

**Comment by linminhtoo** [comment]
> Hi, I cannot reproduce your issue. Both the Fortran implementation and the Python API give the same results for me. I used version 0.3.0 for both and Python 3.9.
> 
> `tblite run --method gfn1 coord`
> `Python API`
> Considering the energy is way too high in the first step, this is not a convergence issue. Have you tested the Fortran version?

hey @marvinfriede , this is really weird... could you share the exact code you're running for the `Python API`? and how did you install `tblite`?

**Comment by marvinfriede** [comment]
I just copied your code and added the import that was missing in your example:
```python
from tblite.interface import Calculator

calc = Calculator(...)
```

I installed `tblite-python` from conda (`mamba install tblite-python`).

**Comment by linminhtoo** [comment]
> I just copied your code and added the import that was missing in your example:
> ```python
> from tblite.interface import Calculator
> 
> calc = Calculator(...)
> ```
> 
> I installed `tblite-python` from conda (`mamba install tblite-python`).

Thanks a lot marvin, I'm trying to pinpoint the issue. As I'm using a different OS (Nix-OS), it might have something to do with it

**Comment by linminhtoo** [comment]
hi @marvinfriede  and others, I spent some time trying different ways of fetching/building the python bindings of `tblite`. after some trial and error, I was able to get it to work (it's mostly due to intricacies of `Nix-OS` and how people have packaged `tblite` in `Nix`). indeed, the version I fetch with `micromamba install tblite-python` works. 

now, the convergence rate is much better. i got a 99% convergence rate on some sample molecules and protein polypeptides. i'll make a new issue on the 1% that didn't converge. 

thanks again.
